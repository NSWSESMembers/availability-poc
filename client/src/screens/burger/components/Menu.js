import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { FlatList } from 'react-native';

import { Container } from '../../../components/Container';
import { ListItem } from '../../../components/List';

class Menu extends Component {
  renderItem = ({ item }) => (
    <ListItem
      bold
      {...item}
    />
  )

  render() {
    const { props } = this;
    const items = [
      {
        title: 'User Profile',
        subtitle: 'Modify your display name and preferences',
        iconLeft: 'user-circle',
        onPress: props.onShowUserProfile,
      },
      {
        title: 'Force location update',
        subtitle: 'Test the location tracking system',
        iconLeft: 'compass',
        onPress: props.onUpdateLocation,
      },
      {
        title: 'Check for updates',
        subtitle: 'Install a new version of the app, if available',
        iconLeft: 'download',
        onPress: props.onCheckForUpdate,
      },
      {
        title: 'Submit feedback',
        subtitle: 'Please post in the Facebook feedback group',
        iconLeft: 'comments',
        onPress: props.onSubmitFeedback,
      },
      {
        title: 'Test Bugsnag',
        subtitle: 'Submit a mock bug report to Bugsnag',
        iconLeft: 'bug',
        onPress: props.onTestBugsnag,
      },
      {
        title: 'Internal Parameters',
        subtitle: 'Display the app version and other details',
        iconLeft: 'info-circle',
        onPress: props.onShowParams,
      },
      {
        title: 'Logout',
        subtitle: 'Sign out of this device',
        iconLeft: 'times-circle',
        onPress: props.onLogout,
      },
    ];

    return (
      <Container>
        <FlatList
          data={items}
          keyExtractor={item => item.title}
          renderItem={this.renderItem}
        />
      </Container>
    );
  }
}

Menu.propTypes = {
  onShowUserProfile: PropTypes.func.isRequired,
  onUpdateLocation: PropTypes.func.isRequired,
  onCheckForUpdate: PropTypes.func.isRequired,
  onSubmitFeedback: PropTypes.func.isRequired,
  onTestBugsnag: PropTypes.func.isRequired,
  onShowParams: PropTypes.func.isRequired,
  onLogout: PropTypes.func.isRequired,
};

export default Menu;
