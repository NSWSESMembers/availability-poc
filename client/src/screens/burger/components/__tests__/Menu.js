import React from 'react';
import TestRenderer from 'react-test-renderer';

import Menu from '../Menu';

test('renders correctly', () => {
  const noop = jest.fn();
  const tree = TestRenderer.create(
    <Menu
      onShowUserProfile={noop}
      onUpdateLocation={noop}
      onCheckForUpdate={noop}
      onSubmitFeedback={noop}
      onTestBugsnag={noop}
      onShowParams={noop}
      onSendTestPush={noop}
      onLogout={noop}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
