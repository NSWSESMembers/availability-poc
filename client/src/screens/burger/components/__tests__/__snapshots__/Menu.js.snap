// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#EEE",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
    ]
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "iconLeft": "user-circle",
          "onPress": [Function],
          "subtitle": "Modify your display name and preferences",
          "title": "User Profile",
        },
        Object {
          "iconLeft": "compass",
          "onPress": [Function],
          "subtitle": "Test the location tracking system",
          "title": "Force location update",
        },
        Object {
          "iconLeft": "download",
          "onPress": [Function],
          "subtitle": "Install a new version of the app, if available",
          "title": "Check for updates",
        },
        Object {
          "iconLeft": "comments",
          "onPress": [Function],
          "subtitle": "Please post in the Facebook feedback group",
          "title": "Submit feedback",
        },
        Object {
          "iconLeft": "bug",
          "onPress": [Function],
          "subtitle": "Submit a mock bug report to Bugsnag",
          "title": "Test Bugsnag",
        },
        Object {
          "iconLeft": "download",
          "onPress": [Function],
          "subtitle": "Send a notification to this device",
          "title": "Send Test Push Notification",
        },
        Object {
          "iconLeft": "info-circle",
          "onPress": [Function],
          "subtitle": "Display the app version and other details",
          "title": "Internal Parameters",
        },
        Object {
          "iconLeft": "times-circle",
          "onPress": [Function],
          "subtitle": "Sign out of this device",
          "title": "Logout",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 2,
                  "elevation": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 0,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 6,
                  "paddingBottom": 10,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 10,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingRight": 8,
                  "width": 40,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 30,
                    },
                    Object {
                      "alignSelf": "center",
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Object {
                    "flex": 0,
                    "fontWeight": "bold",
                    "includeFontPadding": false,
                    "marginRight": 8,
                    "textAlignVertical": "top",
                  }
                }
              >
                User Profile
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Object {
                    "color": "#6B6B6B",
                    "flex": 0,
                    "fontSize": 12,
                    "includeFontPadding": false,
                    "textAlignVertical": "bottom",
                  }
                }
              >
                Modify your display name and preferences
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 2,
                  "elevation": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 0,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 6,
                  "paddingBottom": 10,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 10,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingRight": 8,
                  "width": 40,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 30,
                    },
                    Object {
                      "alignSelf": "center",
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Object {
                    "flex": 0,
                    "fontWeight": "bold",
                    "includeFontPadding": false,
                    "marginRight": 8,
                    "textAlignVertical": "top",
                  }
                }
              >
                Force location update
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Object {
                    "color": "#6B6B6B",
                    "flex": 0,
                    "fontSize": 12,
                    "includeFontPadding": false,
                    "textAlignVertical": "bottom",
                  }
                }
              >
                Test the location tracking system
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 2,
                  "elevation": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 0,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 6,
                  "paddingBottom": 10,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 10,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingRight": 8,
                  "width": 40,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 30,
                    },
                    Object {
                      "alignSelf": "center",
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Object {
                    "flex": 0,
                    "fontWeight": "bold",
                    "includeFontPadding": false,
                    "marginRight": 8,
                    "textAlignVertical": "top",
                  }
                }
              >
                Check for updates
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Object {
                    "color": "#6B6B6B",
                    "flex": 0,
                    "fontSize": 12,
                    "includeFontPadding": false,
                    "textAlignVertical": "bottom",
                  }
                }
              >
                Install a new version of the app, if available
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 2,
                  "elevation": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 0,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 6,
                  "paddingBottom": 10,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 10,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingRight": 8,
                  "width": 40,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 30,
                    },
                    Object {
                      "alignSelf": "center",
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Object {
                    "flex": 0,
                    "fontWeight": "bold",
                    "includeFontPadding": false,
                    "marginRight": 8,
                    "textAlignVertical": "top",
                  }
                }
              >
                Submit feedback
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Object {
                    "color": "#6B6B6B",
                    "flex": 0,
                    "fontSize": 12,
                    "includeFontPadding": false,
                    "textAlignVertical": "bottom",
                  }
                }
              >
                Please post in the Facebook feedback group
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 2,
                  "elevation": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 0,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 6,
                  "paddingBottom": 10,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 10,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingRight": 8,
                  "width": 40,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 30,
                    },
                    Object {
                      "alignSelf": "center",
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Object {
                    "flex": 0,
                    "fontWeight": "bold",
                    "includeFontPadding": false,
                    "marginRight": 8,
                    "textAlignVertical": "top",
                  }
                }
              >
                Test Bugsnag
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Object {
                    "color": "#6B6B6B",
                    "flex": 0,
                    "fontSize": 12,
                    "includeFontPadding": false,
                    "textAlignVertical": "bottom",
                  }
                }
              >
                Submit a mock bug report to Bugsnag
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 2,
                  "elevation": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 0,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 6,
                  "paddingBottom": 10,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 10,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingRight": 8,
                  "width": 40,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 30,
                    },
                    Object {
                      "alignSelf": "center",
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Object {
                    "flex": 0,
                    "fontWeight": "bold",
                    "includeFontPadding": false,
                    "marginRight": 8,
                    "textAlignVertical": "top",
                  }
                }
              >
                Send Test Push Notification
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Object {
                    "color": "#6B6B6B",
                    "flex": 0,
                    "fontSize": 12,
                    "includeFontPadding": false,
                    "textAlignVertical": "bottom",
                  }
                }
              >
                Send a notification to this device
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 2,
                  "elevation": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 0,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 6,
                  "paddingBottom": 10,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 10,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingRight": 8,
                  "width": 40,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 30,
                    },
                    Object {
                      "alignSelf": "center",
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Object {
                    "flex": 0,
                    "fontWeight": "bold",
                    "includeFontPadding": false,
                    "marginRight": 8,
                    "textAlignVertical": "top",
                  }
                }
              >
                Internal Parameters
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Object {
                    "color": "#6B6B6B",
                    "flex": 0,
                    "fontSize": 12,
                    "includeFontPadding": false,
                    "textAlignVertical": "bottom",
                  }
                }
              >
                Display the app version and other details
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 2,
                  "elevation": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 0,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 6,
                  "paddingBottom": 10,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 10,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingRight": 8,
                  "width": 40,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 30,
                    },
                    Object {
                      "alignSelf": "center",
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Object {
                    "flex": 0,
                    "fontWeight": "bold",
                    "includeFontPadding": false,
                    "marginRight": 8,
                    "textAlignVertical": "top",
                  }
                }
              >
                Logout
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Object {
                    "color": "#6B6B6B",
                    "flex": 0,
                    "fontSize": 12,
                    "includeFontPadding": false,
                    "textAlignVertical": "bottom",
                  }
                }
              >
                Sign out of this device
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
